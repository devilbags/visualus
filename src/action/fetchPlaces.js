// @flow

import {
  FETCH_PLACES_BEGIN,
  FETCH_PLACES_COMPLETE,
  FETCH_PLACES_FAIL
} from '../actionType/fetchPlacesActionType';

export type BeginFetchPlaces = {
  type: string,
  payload: {
    input: string,
    components: string,
    types: string
  }
};
export type CompleteFetchPlaces = {
  type: string,
  data: {
    predictions: Array<mixed>,
    status: string
  }
};
export type FailFetchPlaces = {
  type: string,
  error: string
};

export type FetchPlacesAction =
  | BeginFetchPlaces
  | CompleteFetchPlaces
  | FailFetchPlaces;

export function fetchBegin(payload: {
  input: string,
  components: string,
  types: string
}): BeginFetchPlaces {
  return {
    type: FETCH_PLACES_BEGIN,
    payload
  };
}

export function fetchComplete(data: {
  predictions: Array<mixed>,
  status: string
}): CompleteFetchPlaces {
  return {
    type: FETCH_PLACES_COMPLETE,
    data
  };
}

export function fetchFail(error: string): FailFetchPlaces {
  return {
    type: FETCH_PLACES_FAIL,
    error
  };
}
