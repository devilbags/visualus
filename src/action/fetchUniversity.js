// @flow

import {
  FETCH_UNIVERSITY_NAME_BEGIN,
  FETCH_UNIVERSITY_NAME_COMPLETE,
  FETCH_UNIVERSITY_NAME_FAIL
} from '../actionType/fetchUniversityActionType';

type BeginFetchUniversityNames = {
  type: string,
  payload: {
    input: string
  }
};
type CompleteFetchUniversityNames = {
  type: string,
  data: Array<{
    id: number,
    name: string,
    alias: string
  }>
};
type FailFetchUniversity = {
  type: string,
  error: string
};

export type fetchUniversityAction =
  | BeginFetchUniversityNames
  | CompleteFetchUniversityNames
  | FailFetchUniversity;

export function fetchUniversityBegin(payload: {
  input: string
}): BeginFetchUniversityNames {
  return {
    type: FETCH_UNIVERSITY_NAME_BEGIN,
    payload
  };
}

export function fetchUniversityComplete(
  data: Array<{
    id: number,
    name: string,
    alias: string
  }>
): CompleteFetchUniversityNames {
  return {
    type: FETCH_UNIVERSITY_NAME_COMPLETE,
    data
  };
}

export function fetchUniversityFail(error: string): FailFetchUniversity {
  return {
    type: FETCH_UNIVERSITY_NAME_FAIL,
    error
  };
}
