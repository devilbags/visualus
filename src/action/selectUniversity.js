// @flow

import type { Id, UniversityActions } from '../type/university';

export const setSelectUniversity = (id: Id): UniversityActions => {
  return {
    type: 'SET_SELECTED_UNIVERSITY',
    id: id
  };
};
