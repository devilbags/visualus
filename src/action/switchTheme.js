import { SWITCH_THEME } from '../actionType/switchThemeActionType';

const switchTheme = () => {
  return {
    type: SWITCH_THEME
  };
};

export default switchTheme;
