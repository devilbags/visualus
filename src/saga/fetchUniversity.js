import { call, takeLatest, put } from 'redux-saga/effects';
import { universityList } from '../api/eduData';

export function* fetchUniversity(action) {
  try {
    const result = yield call(universityList, action.input);
    yield put({ type: 'FETCH_UNIVERSITY_COMPLETE', result: result });
  } catch (error) {
    yield put({ type: 'FETCH_UNIVERSITY_FAIL', error: error });
  }
}

export default function* watchFetchUniversity() {
  yield takeLatest('FETCH_UNIVERSITY_BEGIN', fetchUniversity);
}
