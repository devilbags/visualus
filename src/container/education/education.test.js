import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import Education from './education';
import { beginFetchUniversity } from '../../action/fetchUniversity';
import { setSelectUniversity } from '../../action/selectUniversity';

const setup = (setupProps = {}) => {
  const store = configureStore()({
    Education: {
      loading: false,
      metaData: {
        total: 0,
        per_page: 20,
        page: 0
      },
      keyword: '',

      universityList: [
        {
          id: 1,
          name: 'First Unviersity',
          alias: 'FU'
        },
        {
          id: 2,
          name: 'Second University',
          alias: 'SU'
        },
        {
          id: 3,
          name: 'Third University',
          alias: 'TU'
        }
      ]
    }
  });
  store.dispatch = jest.fn();
  const wrapper = shallow(<Education store={store} />);
  return {
    store,
    wrapper
  };
};

describe('Education container', () => {
  it('Should render without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('Should dispatch begin fetch action', () => {
    const { wrapper, store } = setup();
    wrapper.props().onInputValueChange('test');
    expect(store.dispatch).toHaveBeenCalledWith(
      beginFetchUniversity('test', 0, 20)
    );
  });
  it('Should dispatch set select university action', () => {
    const { wrapper, store } = setup();
    wrapper.props().onUniversitySelected(1);
    expect(store.dispatch).toHaveBeenCalledWith(setSelectUniversity(1));
  });
});
