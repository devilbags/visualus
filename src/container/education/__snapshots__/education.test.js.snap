// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education container Should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Education)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "loading": false,
      "onInputValueChange": [Function],
      "onUniversitySelected": [Function],
      "selectedUniversity": null,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "universityList": Array [
        Object {
          "alias": "FU",
          "id": 1,
          "name": "First Unviersity",
        },
        Object {
          "alias": "SU",
          "id": 2,
          "name": "Second University",
        },
        Object {
          "alias": "TU",
          "id": 3,
          "name": "Third University",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "loading": false,
        "onInputValueChange": [Function],
        "onUniversitySelected": [Function],
        "selectedUniversity": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "universityList": Array [
          Object {
            "alias": "FU",
            "id": 1,
            "name": "First Unviersity",
          },
          Object {
            "alias": "SU",
            "id": 2,
            "name": "Second University",
          },
          Object {
            "alias": "TU",
            "id": 3,
            "name": "Third University",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
